test/test_video.py

```diff
@@ -0,0 +1,407 @@
+
+    if e is None:
+        e = float("inf")
+    if e < s:
+        raise ValueError(
+            "end time should be larger than start time, got "
+            "start time={} and end time={}".format(s, e)
+        )
+    video_object.set_current_stream("video")
+    video_object.seek(s)
+    video_frames = torch.empty(0)
+    frames = []
+    video_pts = []
+    t, pts = video_object.next()
+    while t.numel() > 0 and (pts >= s and pts <= e):
+        frames.append(t)
+        video_pts.append(pts)
+        t, pts = video_object.next()
+    if len(frames) > 0:
+        video_frames = torch.stack(frames, 0)
+
+    video_object.set_current_stream("audio")
+    video_object.seek(s)
+    audio_frames = torch.empty(0)
+    frames = []
+    audio_pts = []
+    t, pts = video_object.next()
+    while t.numel() > 0 and (pts > s and pts <= e):
+        frames.append(t)
+        audio_pts.append(pts)
+        t, pts = video_object.next()
+    if len(frames) > 0:
+        audio_frames = torch.stack(frames, 0)
+
+    return DecoderResult(
+        vframes=video_frames,
+        vframe_pts=video_pts,
+        vtimebase=None,
+        aframes=audio_frames,
+        aframe_pts=audio_pts,
+        atimebase=None,
+    )
+    return video_frames, audio_frames, video_object.get_metadata()
+
+
+@unittest.skipIf(_HAS_VIDEO_OPT is False, "Didn't compile with ffmpeg")
+class TestVideo(unittest.TestCase):
+    @unittest.skipIf(av is None, "PyAV unavailable")
+    def test_read_video_tensor(self):
+        """
+        Check if reading the video using the `next` based API yields the
+        same sized tensors as the pyav alternative.
+        """
+        torchvision.set_video_backend("pyav")
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+            # pass 1: decode all frames using existing TV decoder
+            tv_result, _, _ = torchvision.io.read_video(full_path, pts_unit="sec")
+            tv_result = tv_result.permute(0, 3, 1, 2)
+            # pass 2: decode all frames using new api
+            reader = torch.classes.torchvision.Video(full_path, "video")
+            frames = []
+            t, _ = reader.next()
+            while t.numel() > 0:
+                frames.append(t)
+                t, _ = reader.next()
+            new_api = torch.stack(frames, 0)
+            self.assertEqual(tv_result.size(), new_api.size())
+
+    def test_partial_video_reading_fn(self):
+        torchvision.set_video_backend("video_reader")
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+
+            # select two random points between 0 and duration
+            r = []
+            r.append(random.uniform(0, config.duration))
+            r.append(random.uniform(0, config.duration))
+            s = min(r)
+            e = max(r)
+
+            reader = torch.classes.torchvision.Video(full_path, "video")
```

Comment:
For a follow-up PR: we should expose in torchvision `Video`, so that you can access it via `torchvision.io.Video` or something like that

Reply:
adding that to #2660 feature tracker